### PR TITLE
User/kmadisa/use latest pytango features

### PR DESCRIPTION
**Pull request by kmadisa**
**PyTango >=9.2.2** now supports [dynamic commands](https://github.com/tango-controls/pytango/pull/122). Previously we had to create a class https://github.com/ska-sa/tango-simlib/blob/ec05e9d33950a7428f64c8aaafd4fad41b85a485/tango_simlib/tango_sim_generator.py#L91-L97 and populate its commands' dictionary https://github.com/ska-sa/tango-simlib/blob/ec05e9d33950a7428f64c8aaafd4fad41b85a485/tango_simlib/tango_sim_generator.py#L235-L245
 before we started the device. However with this new feature we are able to add new commands to the device at runtime.

### CODE REVIEW

File: tango_simlib/sim_test_interface.py [hunk 1]
```diff
@@ -43,7 +43,16 @@ def init_device(self):
                                'correct value.'.format(self.model_key))
         self.sim_device_attributes = self.model.sim_quantities.keys()
         self.set_state(DevState.ON)
-
+        self.initialize_dynamic_commands()
+        
+
```
Comment (ajoubertza) [inline tango_simlib/sim_test_interface.py]: Please remove the extra blank line between the methods (pep8)

File: tango_simlib/tango_sim_generator.py [hunk 2]
```diff
@@ -253,6 +212,8 @@ def init_device(self):
             self._not_added_attributes = []
             write_device_properties_to_db(self.get_name(), self.model)
             self._reset_to_default_state()
+            self.initialize_dynamic_commands()
+
```
Comment (ajoubertza) [inline tango_simlib/tango_sim_generator.py]: Extra line here too.

File: tango_simlib/utilities/helper_module.py [hunk 3]
```diff
@@ -55,6 +58,13 @@
 
 TANGO_NOT_SPECIFIED_PROPS = ['Not specified', 'No display unit', 'No standard unit']
 
+POGO_USER_DEFAULT_CMD_PROP_MAP = {
+        'name': 'name',
+        'arginDescription': 'doc_in',
+        'arginType': 'dtype_in',
+        'argoutDescription': 'doc_out',
+        'argoutType': 'dtype_out'}
```
Comment (ajoubertza) [inline tango_simlib/utilities/helper_module.py]: The keys in this dict never get used, so do we need a dict here, or just a tuple?  Also, the old copy in `tango_sim_generator.py` can be dropped now.  There is also one in `sim_xmi_parser.py`, which just has one extra key - `inherited`.  Can share one copy everywhere?

File: tango_simlib/utilities/helper_module.py [hunk 4]
```diff
@@ -105,6 +115,27 @@ def get_host_address():
...
+            return action_handler(tango_dev=tango_device, data_input=input_parameters)
+
+        cmd_handler.__name__ = action_name
+        cmd_info_copy = model.sim_actions_meta[action_name].copy()
+        # Delete all the keys that are not part of the Tango command parameters.
+        cmd_info_copy.pop('name')
+        tango_cmd_prop = POGO_USER_DEFAULT_CMD_PROP_MAP.values()
+        for prop_key in model.sim_actions_meta[action_name]:
+            if prop_key not in tango_cmd_prop:
+                MODULE_LOGGER.info(
```
Comment (ajoubertza) [inline tango_simlib/utilities/helper_module.py]: Please change this log to `warning` level.

File: tango_simlib/tango_sim_generator.py [hunk 5]
```diff
@@ -72,7 +72,7 @@ def read_attributes(self, attr):
 
 def get_tango_device_server(model, sim_data_files):
     """Declares a tango device class that inherits the Device class and then
-    adds tango commands.
+    adds tango attributes (DevEnum and Specrum type).
```
Comment (lvdheever) [inline tango_simlib/tango_sim_generator.py]: Typo: Spectrum

File: tango_simlib/utilities/helper_module.py [hunk 6]
```diff
@@ -123,10 +118,10 @@ def cmd_handler(tango_device, input_parameters=None):
         cmd_info_copy = model.sim_actions_meta[action_name].copy()
         # Delete all the keys that are not part of the Tango command parameters.
         cmd_info_copy.pop('name')
-        tango_cmd_prop = POGO_USER_DEFAULT_CMD_PROP_MAP.values()
+        tango_cmd_props = DEFAULT_CMD_PROPS
         for prop_key in model.sim_actions_meta[action_name]:
-            if prop_key not in tango_cmd_prop:
-                MODULE_LOGGER.info(
+            if prop_key not in tango_cmd_props:
```
Comment (ajoubertza) [inline tango_simlib/utilities/helper_module.py]: Could just iterate through `DEFAULT_CMD_PROPS` instead of assigning another variable, but no matter.

**Pull request merged**